kolibri/plugins/device_management/assets/src/views/DeviceSettingsPage/index.vue

```diff
@@ -0,0 +1,149 @@
+      </p>
+    </section>
+
+    <section>
+      <UiAlert
+        v-if="saveStatus === 'SUCCESS'"
+        type="success"
+        @dismiss="resetSaveStatus"
+      >
+        {{ $tr('saveSuccessNotification') }}
+      </UiAlert>
+      <UiAlert
+        v-if="saveStatus === 'ERROR'"
+        type="error"
+        @dismiss="resetSaveStatus"
+      >
+        {{ $tr('saveFailureNotification') }}
+      </UiAlert>
+    </section>
+    <section>
+      <h3>
+        {{ $tr('defaultLanguageHeader') }}
+      </h3>
+      <KSelect
+        v-model="language"
+        :label="$tr('selectedLanguageLabel')"
+        :options="languageOptions"
+        :disabled="language.value === undefined"
+        :floatingLabel="false"
+      />
+    </section>
+
+    <section>
+      <KButton
+        class="save-button"
+        :text="$tr('saveAction')"
+        appearance="raised-button"
+        primary
+        @click="handleClickSave"
+      />
+    </section>
+  </div>
+
+</template>
+
+
+<script>
+
+  import mapValues from 'lodash/map';
+  import find from 'lodash/find';
+  import KButton from 'kolibri.coreVue.components.KButton';
+  import KSelect from 'kolibri.coreVue.components.KSelect';
+  import UiAlert from 'keen-ui/src/UiAlert';
+  import { availableLanguages } from 'kolibri.utils.i18n';
+  import { getDeviceLanguageSetting, saveDeviceLanguageSetting } from './api';
+
+  export default {
+    name: 'DeviceSettingsPage',
+    components: {
+      KButton,
+      KSelect,
+      UiAlert,
+    },
+    metaInfo() {
+      return {
+        title: this.$tr('pageHeader'),
+      };
+    },
+    data() {
+      return {
+        language: {},
+        saveStatus: null,
+        browserDefaultOption: {
+          value: null,
+          label: this.$tr('browserDefaultLanguage'),
+        },
+      };
+    },
+    computed: {
+      languageOptions() {
+        return [
+          this.browserDefaultOption,
+          ...mapValues(availableLanguages, language => {
+            return {
+              value: language.id,
+              label: language.lang_name,
+            };
+          }),
+        ];
+      },
+    },
+    beforeMount() {
+      this.getDeviceLanguageSetting().then(languageId => {
+        const match = find(this.languageOptions, { value: languageId });
+        if (match) {
+          this.language = { ...match };
+        } else {
+          this.language = this.browserDefaultOption;
+        }
+      });
+    },
+    methods: {
+      resetSaveStatus() {
+        this.saveStatus = null;
+      },
+      handleClickSave() {
+        this.resetSaveStatus();
+        this.saveDeviceLanguageSetting(this.language.value)
+          .then(() => {
+            this.saveStatus = 'SUCCESS';
+          })
+          .catch(() => {
+            this.saveStatus = 'ERROR';
+          });
+      },
+      getDeviceLanguageSetting,
+      saveDeviceLanguageSetting,
+    },
+    $trs: {
+      browserDefaultLanguage: 'Browser default',
```

Comment:
Super!

Reply:
Oh - we should add this option to the setup wizard language picker also.

Reply:
Good point!

Reply:
The setup wizard uses the same language picker on the sign-on page. Should we add "Browser default" to all the language selectors, including the modal type?

Reply:
Making this work via the language cookie for individual users is probably beyond the scope of the feature, and might be a little more complicated to implement.

To add this in the setup wizard, we will probably need to add a special flag to provide "Browser default" as an option to the LanguageSwitcherList.

Reply:
We could let the language switcher expose this option, but I think it might be more confusing to users, so let's not give the option to clear the cookie until people ask for it.

> To add this in the setup wizard, we will probably need to add a special flag to provide "Browser default" as an option to the LanguageSwitcherList.

I think this is the probably what we need to do.

Reply:
Actually, this might be messy too. Since we'd need to update the modal anyway.

How about we just put a button below the language list on Step 1 of the Setup Wizard to the effect of "Use the browser default"?

Reply:
Or can we just consider this PR done and punt the Setup Wizard stuff to a new feature?